### PR TITLE
Add check for correct signature

### PR DESCRIPTION
**Pull request by rjaccard**

### CODE REVIEW

File: be1-go/hub/organizer/mod_test.go [hunk 1]
```diff
@@ -460,12 +462,11 @@ type keypair struct {
 }
 
 var nolog = zerolog.New(io.Discard)
-var suite = crypto.Suite
+//var suite = crypto.Suite
```
Comment (Anghul) [inline be1-go/hub/organizer/mod_test.go:465]: Does it still work if you remove the variable ? It seems to be used in generateKeyPair

File: be1-go/hub/organizer/mod.go [hunk 2]
```diff
@@ -291,6 +295,30 @@ func (h *Hub) handleMessageFromClient(incomingMessage *socket.IncomingMessage) e
 	return nil
 }
 
+func verifySignature(publicKey64 string, signature64 string, data64 string) error {
+	pk, err := base64.URLEncoding.DecodeString(publicKey64)
+	if err != nil {
+		return xerrors.Errorf("failed to decode the public key")
```
Comment (Anghul) [inline be1-go/hub/organizer/mod.go:301]: It could be a good idea to return err too, and same comment for the returns below

**Review by Anghul — CHANGES_REQUESTED** [review]
Good job overall, just a few errors that could be returned to have more complete logs

File: be1-go/hub/organizer/mod_test.go (hunk 1 above)
Comment (rjaccard) [reply be1-go/hub/organizer/mod_test.go:465]: I moved it to mod.go

File: be1-go/hub/organizer/mod.go (hunk 2 above)
Comment (rjaccard) [reply be1-go/hub/organizer/mod.go:301]: nice catch!

**Review by Anghul — APPROVED** [review]

**Review by celinecamacho — APPROVED** [review]
LGTM

File: be1-go/hub/organizer/mod.go [hunk 3]
```diff
@@ -291,6 +295,30 @@ func (h *Hub) handleMessageFromClient(incomingMessage *socket.IncomingMessage) e
...
+	}
```
Comment (nkcr) [inline be1-go/hub/organizer/mod.go:302]: missing return

File: be1-go/hub/organizer/mod.go [hunk 4]
```diff
@@ -291,6 +295,30 @@ func (h *Hub) handleMessageFromClient(incomingMessage *socket.IncomingMessage) e
...
+	if err != nil {
+		return xerrors.Errorf("failed to decode the signature: %v", err)
+	}
+
+	data, err := base64.URLEncoding.DecodeString(data64)
+	if err != nil {
+		return xerrors.Errorf("failed to decode the data: %v", err)
+	}
+
+	return schnorr.VerifyWithChecks(crypto.Suite, pk, data, signature)
```
Comment (nkcr) [inline be1-go/hub/organizer/mod.go:319]: I'm not a big fan of returning an invocation of a function. It makes the code harder to follow and can make debugging harder, with less meaningful error message.

**Review by nkcr — APPROVED** [review]

**Pull request merged**